### PR TITLE
Feature/plot modular

### PR DESCRIPTION
**Pull request by j-ti**
in the PR I just seperated the extraction of results from the actual plotting,
(added) outsourcing the labels and titles into dictionaries
(added) failing test: adept example model attribute

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -4,4 +4,4 @@ pandas
 numpy
 deap
 oemof
-scipy
\ No newline at end of file
+scipy
```
Comment (stefansc1) [inline requirements.txt:7]: Where is this imported?

File: smooth/framework/functions/functions.py [hunk 2]
```diff
@@ -42,3 +42,119 @@ def get_sim_time_span(n_interval, step_size):
...
+                nb_intervals = len(component_flows[flow])
+                nb_trailing_none = nb_intervals
+                for flow_val in component_flows[flow]:
+                    if flow_val is not None:
+                        nb_trailing_none -= 1
+                    else:
+                        break
+                # check if it's a chp component which consists of two oemof models
+                # if so get rid of the ending '_electric' or '_thermal'
+                if this_flow_name_split[0][-9:] == '_electric':
```
Comment (stefansc1) [inline smooth/framework/functions/functions.py:72]: Have a look at the endswith() function of strings.

File: smooth/framework/functions/functions.py [hunk 3]
```diff
@@ -42,3 +42,119 @@ def get_sim_time_span(n_interval, step_size):
...
+                            updated_bus_list.append(this_val)
+                        # Override the old bus list with the updated one.
+                        this_comp_flows[bus] = updated_bus_list
+                    else:
+                        # Case: Component has no flow with this bus yet.
+                        this_comp_flows[bus] = [-this_val for this_val in
+                                                component_flows[flow][:nb_intervals - nb_trailing_none]]
+
+            for this_bus in this_comp_flows:
+                # Replaces shorthand component names in the results with the official names for those listed.
```
Comment (stefansc1) [inline smooth/framework/functions/functions.py:110]: Might be done in dictionary. Translating strings is the literal definition of a dictionary.

File: smooth/framework/functions/functions.py (hunk 2 above)
Comment (j-ti) [reply smooth/framework/functions/functions.py:72]: I did not change the function, as in the discription said, its only about exporting the functionality not about optimizing it

File: requirements.txt (hunk 1 above)
Comment (j-ti) [reply requirements.txt:7]: https://github.com/rl-institut/smooth/blob/dev/smooth/components/component_functions/all_component_functions.py

File: smooth/framework/functions/functions.py (hunk 2 above)
Comment (j-ti) [reply smooth/framework/functions/functions.py]: https://github.com/rl-institut/smooth/pull/77/commits/6f8f2544e909f93210dd6a3a4505b1174a4b3488

File: smooth/framework/functions/functions.py (hunk 3 above)
Comment (j-ti) [reply smooth/framework/functions/functions.py:118]: Already have an issue #79 for that
It still does not make sense to have a dictionary hard coded, as the user can freely choose the names. So it has to be done somewhere else anyways

Comment (j-ti) [reply smooth/framework/functions/functions.py]: Came up with an optional argument label dict for the plotting function, this way it is not hard coded anymore
https://github.com/rl-institut/smooth/pull/77/commits/5394241d8718615e498fb74331dab14a193ad4a3

**Pull request merged**